src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java

```diff
@@ -74,9 +73,8 @@ protected void retryableGetMlTask(
         String workflowStep,
         ActionListener<MLTask> mlTaskListener
     ) {
-        AtomicInteger retries = new AtomicInteger();
         CompletableFuture.runAsync(() -> {
-            while (retries.getAndIncrement() < this.maxRetry && !future.isDone()) {
+            while (!future.isDone()) {
```

Comment:
I'm probably glancing over it but where do we take into account the give timeout for each step to stop and call the timeout now

Reply:
The `ProcessNode` handles the step timeouts:
https://github.com/opensearch-project/flow-framework/blob/e0f5a2a1a95c794d8a899e5aa052725871cd9a88/src/main/java/org/opensearch/flowframework/workflow/ProcessNode.java#L164-L170

Basically it starts the timeout thread, scheduled to complete exceptionally at the timeout.

Then it starts the step execute thread.

Whichever thread completes the future first wins.... if the step completes, it cancels the delay thread (if for some reason it doesn't do so fast enough, the future can only be completed once so the complete exceptionally becomes a no-op).

Reply:
I see, thanks for pointing to this